tests/data/attributes/test_dict.py

```diff
@@ -38,14 +39,15 @@ def test_default_init(self):
     )
     def test_value_init(self, value):
         """Test that a DatasetDict is correctly value-initialized."""
-        dset_dict = DatasetDict(value)
+        with np.printoptions(legacy="1.21"):
+            dset_dict = DatasetDict(value)
 
-        assert dset_dict == value
-        assert dset_dict.info.type_id == "dict"
-        assert dset_dict.info.py_type == "dict"
-        assert dset_dict.bind.keys() == value.keys()
-        assert len(dset_dict) == len(value)
-        assert repr(value) == repr(dset_dict)
+            assert dset_dict == value
+            assert dset_dict.info.type_id == "dict"
+            assert dset_dict.info.py_type == "dict"
+            assert dset_dict.bind.keys() == value.keys()
+            assert len(dset_dict) == len(value)
+            assert repr(value) == repr(dset_dict)
```

Comment:
If `dset_dict` is set inside the context manager, we shouldn't have to put these assertions in the context manager as well.

Reply:
Same comment for other assertions as well.